CoffeeMachine_Task1/src/bg/coffee_machine/PremiumCoffeeMachine.java

```diff
@@ -0,0 +1,99 @@
+package bg.coffee_machine;
+
+public class PremiumCoffeeMachine implements CoffeeMachine {
+    private PremiumContainer container;
+    private boolean autoRefill = false;
+    private final String[] allowedLucks = {
+            "If at first you don't succeed call it version 1.0.",
+            "Today you will make magic happen!",
+            "Have you tried turning it off and on again?",
+            "Life would be much more easier if you had the source code.",
+    };
+
+    public PremiumCoffeeMachine(boolean autoRefill) {  this(); this.autoRefill = autoRefill; }
+    public PremiumCoffeeMachine() {
+        this.container = new PremiumContainer();
+    }
+
+    @Override
+    public Product brew(Beverage beverage) {
+        if(beverage instanceof Espresso)
+            if(enoughEspresso(beverage)){
+                decreaseSuppliesForEspresso(beverage);
+                return new Product(beverage, allowedLucks);
```

Comment:
Provide only one luck to a product. If there are no lucks (BasicCoffeeMachine) then set null for the product's luck. Iterate the lucks in the PremiumMachine, otherwise always the same luck is returned for every new coffee ordered.